src/layman/publication_relation/util.py

```diff
@@ -7,3 +7,28 @@ def update_related_publications_after_change(workspace, publication_type, public
         maps = get_publication_info(workspace, publication_type, publication, context={'keys': ['used_in_maps']})['used_in_maps']
         for map_obj in maps:
             patch_after_feature_change(map_obj['workspace'], MAP_TYPE, map_obj['name'])
+
+
+def check_no_internal_workspace_name_layer(map_json, *, x_forwarded_items):
+    from layman import names, LaymanError
+    from layman.map.util import get_internal_gs_layers_from_json
+    from layman.layer.prime_db_schema.table import get_layer_info
```

Comment:
I think we prefer using layman.util.get_publication_info over specific source methods.